src/sys/socket/addr.rs

```diff
@@ -64,6 +64,24 @@ impl InetAddr {
         }
     }
 
+    /// Creates an `IpAddr` struct from libc's sockaddr.
+    ///
+    /// Note that this supports only IPv4 and IPv6 address families.
+    pub unsafe fn from_libc_sockaddr(addr: *mut libc::sockaddr) -> Option<InetAddr> {
```

Comment:
I marked this one unsafe because it takes a raw pointer, which could point anywhere and yield a segfault. Again, this is not a public function, but only used internally (by that other `InterfaceAddress::from_libc_ifaddrs` function)